source/docs/software/advanced-controls/introduction/tuning-pid-controller.rst

```diff
@@ -1,32 +1,229 @@
+    <div class="viz-div">
+      <div class="col" id="arm_pid_plot"></div>
+      <div class="flex-grid">
+         <div class="col" id="arm_pid_viz"></div> 
+         <div id="arm_pid_ctrls"></div> 
+      </div>
+      <script>
+         arm_pidf = new VerticalArmPIDF("arm_pid"); 
+         arm_pidf.runSim(); 
+      </script> 
+    </div>
+
+Mechanism Description
+~~~~~~~~~~~~~~~~~~~~~
+The "Vertical Arm" is a mass on a stick, moved up and down by a gearbox, and driven by a motor. They're commonly used to lift gamepieces from the ground, and up higher to place and score them.
+
+Applying voltage to the motor causes a force on the mechanism that drives the arm up or down. If there is no voltage, gravity still acts on the arm to pull it downward.
+
+To consistently place a gamepiece, the arm must move from wherever it is at, to a specific angle which puts the gamepiece at the right height. 
+
+This design drives the controls goal we will use in this example: Put the correct amount of voltage into the motor to get the arm to a certain angle, and then keep it there.
+
+Gearbox inefficiencies and sensor delay is included in this model.
+
+
+Step 1: Feedback-Only
+~~~~~~~~~~~~~~~~~~~~~
+
+Again, we will first attempt to tune this mechanism with using only feedback terms :math:`K_p`, :math:`K_i`, and :math:`K_d`. 
+
+Perform the following:
+
+1. Set :math:`K_p`, :math:`K_i`, :math:`K_d`, and :math:`K_{cosFF}` to zero.
+2. Increase :math:`K_p` until the :term:`output` starts to oscillate. You likely won't be able to push it much higher.
+3. Increase :math:`K_i` when the :term:`output` gets "stuck" before converging to the :term:`setpoint`.
+4. Increase :math:`K_d` as much as possible without introducing jittering in the :term:`system response`. It should help reduce some of the oscillation.
+
+Note that you will likely have trouble finding a set of tunes that behaves acceptably. If you think you have a set, try adjusting the setpoint to be a bit different. You'll likely see the arm behave very differently for small changes in setpoints.
+
+.. raw:: html
+
+   <details>
+     <summary>Tuning Solution</summary>
+
+
+In this particular example, for a setpoint of 0.1, values of :math:`K_p = 12.0`, :math:`K_i = 6.0`, and :math:`K_d = 3.0` will produce somewhat reasonable results. It won't be great for other setpoints.
+   
+.. raw:: html
+
+   </details> <br>
+
+This is a case where feedback control alone is insufficient to achieve good behavior with the system.
+
+Step 2: Feed-Forward, then FeedBack
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The core reason for this is that gravity pulls on the arm in a :term:`non-linear` fashion. That is to say, the amount of :term:`torque` that gravity exerts on our arm is proportional to the *cosine* of the current angle.
```

Comment:
This isn't quite right - even if gravity pulled on the arm in a linear fashion (e.g. an elevator) you still can't really tune with feedback only.  The reason is that we're missing the system model entirely.  Nonlinearity is incidental here.

Reply:
Gravity for an elevator is actually affine, not linear.

Reply:
I made it more hand wavey so it should be more meaningful to the lay audience.